### PR TITLE
[docker] integrate cookiecutter with docker

### PR DESCRIPTION
**Pull request by zouzias**
Integrate docker with cookiecutter by using an alias on the make command.

To test:

Run
```
cookiecutter https://github.com/Sqooba/cookiecutter-pypackage --checkout feature/cookiecutter_in_containers
```
and follow the interactive menu:

```
You've downloaded /Users/anastasios/.cookiecutters/cookiecutter-pypackage before. Is it okay to delete and re-download it? [yes]: yes
full_name [Audrey Roy Greenfeld]: 
email [divers@sqooba.io]: 
github_username [audreyr]: 
project_name [Python Boilerplate]: 
project_slug [python_boilerplate]: 
project_short_description [Python Boilerplate contains all the boilerplate you need to create a Python package.]: 
pypi_username [audreyr]: 
version [0.1.0]: 
use_docker_to_run_make_commands [y]: 
Select docker_python_image:
1 - docker.sqooba.io/sqooba/ocean-python-app:3.7
2 - docker.sqooba.io/sqooba/ocean-python-app:3.6
Choose from 1, 2 (1, 2) [1]: 
use_pytest [y]: 
use_pypi_deployment_with_travis [n]: 
add_pyup_badge [n]: 
Select command_line_interface:
1 - Click
2 - No command-line interface
Choose from 1, 2 (1, 2) [1]: 
create_author_file [y]: 
Select open_source_license:
1 - Proprietary, Sqooba (Schweiz) AG
2 - MIT license
3 - BSD license
4 - ISC license
5 - Apache Software License 2.0
6 - GNU General Public License v3
7 - Not open source
Choose from 1, 2, 3, 4, 5, 6, 7 (1, 2, 3, 4, 5, 6, 7) [1]: 
```

To enable docker in cookiecutter, follow the instructions of `make enable-docker`:

```
make enable-docker 
To run Cookiecutter in Docker mode
You need to first run 'docker-compose build' and then set the following alias on 'make'
alias make='docker-compose run cookiecutter make'
After that, 'make' commands will run within docker
To remove the alias, type, 'unalias make'
```

### CODE REVIEW

**Comment by tukistukistukis** [comment]
I got this error when I was doing docker-compose build
Screenshot 2019-09-30 at 10 16 34
v

**Comment by zouzias** [comment]
> I got this error when I was doing docker-compose build
> Screenshot 2019-09-30 at 10 16 34
> v

I need to fix this, can you do `docker pull docker.sqooba.io/sqooba/ocean-python-app:3.6` for now.

A quick fix is to add it to the list of instructions in `make enable-docker`.

File: {{cookiecutter.project_slug}}/Makefile [hunk 1]
```diff
@@ -53,6 +53,17 @@ clean-test: ## remove test and coverage artifacts
...
 	flake8 {{ cookiecutter.project_slug }} tests
 
+{%- if cookiecutter.use_docker_to_run_make_commands == 'y' %}
+enable-docker: ## Instructions to run cookiecutter in Docker now
+	@echo "To run Cookiecutter in Docker mode"
+	@echo "You need to install 'docker-compose', have access to docker.sqooba.io registry and then set the following alias on 'make'"
+	@echo "alias make='docker-compose run cookiecutter make'"
+	@echo "After that, 'make' commands will run within docker"
+	@echo "To remove the alias, type, 'unalias make'"
+	docker pull {{ cookiecutter.docker_python_image }}
```
Comment (zouzias) [inline {{cookiecutter.project_slug}}/Makefile]: @tukistukistukis I added the pull on the `make enable-docker` step. Now, it should work smoother.

**Comment by tukistukistukis** [comment]
Great! one question: everytime we need to update the code, do we need to rebuild the container manually again?

**Comment by zouzias** [comment]
> Great! one question: everytime we need to update the code, do we need to rebuild the container manually again?

No, you need to build the container only if you change the `Dockerfile` (say want to try out Python 3.6). 

Everytime the container runs, it "mounts" the current working directory inside the container and run the make command. (see https://github.com/Sqooba/cookiecutter-pypackage/blob/feature/cookiecutter_in_containers/%7B%7Bcookiecutter.project_slug%7D%7D/docker-compose.yml#L10)

**Pull request merged**